### PR TITLE
NAVAND-1025: fix the crash in MapboxVoiceInstructionsPlayer

### PR DESCRIPTION
**Pull request by zeac**
### Description
This PR fixes the possible crash. The reproduce steps are unknown, I guess there is a race between calls to `clear` and focus callback calls.

### CODE REVIEW

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt [hunk 1]
```diff
@@ -92,9 +92,11 @@ class MapboxVoiceInstructionsPlayer @JvmOverloads constructor(
                 abandonFocus()
             }
 
-            val currentAnnouncement = currentPlayCallback.announcement
-            val currentClientCallback = currentPlayCallback.consumer
-            currentClientCallback.accept(currentAnnouncement)
+            if (currentPlayCallback != null) {
```
Comment (dzinad) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt:95]: Can you add the following test to `MapboxVoiceInstructionsPlayerTest`:
```
    @Test
    fun `play is done after clear`() {
        val anyLanguage = Locale.US.language
        val mockedAnnouncement: SpeechAnnouncement = mockk(relaxed = true)
        val mapboxVoiceInstructionsPlayer =
            MapboxVoiceInstructionsPlayer(
                aMockedContext,
                anyLanguage,
                mockedPlayerOptions
            )
        val mockedPlay: SpeechAnnouncement = mockedAnnouncement
        val voiceInstructionsPlayerConsumer =
            mockk<MapboxNavigationConsumer<SpeechAnnouncement>>(relaxed = true)

        val requestSlotCallback = slot<AudioFocusRequestCallback>()
        every {
            mockedAudioFocusDelegate.requestFocus(any(), capture(requestSlotCallback))
        } just Runs
        mapboxVoiceInstructionsPlayer.play(mockedPlay, voiceInstructionsPlayerConsumer)
        mapboxVoiceInstructionsPlayer.clear()

        requestSlotCallback.captured.invoke(true)

        verify(exactly = 0) { voiceInstructionsPlayerConsumer.accept(any()) }
    }
```

Comment (dzinad) [inline libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt:95]: We should also add a changelog entry. Sth like:
```
- Fixed a rare `java.lang.NullPointerException: Attempt to read from field 'en.b dn.m.a' on a null object reference` crash in `PlayCallback.getAnnouncement`. [#6760](https://github.com/mapbox/mapbox-navigation-android/pull/6760)
```
to `mapbox-navigation-android/CHANGELOG.md`.

Comment (zeac) [reply libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt:95]: Sure, done.

Comment (dzinad) [reply libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt:95]: Thanks. I'd wait for Tomasz though, he's the one who's been working with voice instructions lately.

**Comment by Guardiola31337** [comment]
@zeac noting that `Ktlint` failed 👀 

```
> Task :libnavui-voice:ktlint
/root/code/libnavui-voice/src/test/java/com/******/navigation/ui/voice/api/MapboxVoiceInstructionsPlayerTest.kt:662:1: Needless blank line(s) (no-consecutive-blank-lines)

> Task :libnavui-voice:ktlint FAILED
```

**Comment by Guardiola31337** [comment]
@zeac it seems we're good merging here 🚀 

cc @dzinad

**Comment by Guardiola31337** [comment]
@zeac 
> it seems we're good merging here 🚀
>> I'd wait for Tomasz though, he's the one who's been working with voice instructions lately.

Oh well, maybe not https://github.com/mapbox/mapbox-navigation-android/pull/6760#discussion_r1052161401 wdyt @dzinad?

cc @tomaszrybakiewicz

File: libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt (hunk 1 above)
Comment (dzinad) [reply libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/MapboxVoiceInstructionsPlayer.kt:95]: Oh, we's out until after the holidays, let's merge.

**Review by dzinad — APPROVED** [review]

**Pull request merged**